modules/nw-ovn-k-anp-best-practices.adoc

```diff
@@ -0,0 +1,61 @@
+
+* Administrators should build ANP that apply to user namespaces not system namespaces.
+
+[IMPORTANT]
+====
+Applying ANP and BANP to system namespaces (`default`, `kube-system`, any namespace whose name starts with `openshift-`, etc) is not supported, and this can leave your cluster unresponsive and in a non-functional state.
+====
+
+* Since `0-100` is the supported priority range, it is always better when designing your ANP to use a middle range like `30-70` because this leaves some placeholder
+for priorities before and after. Even in the middle range, you might want to leave gaps so that as your infrastructure requirements evolve over time, you are able to insert new ANPs when needed at the right priority level. If you pack your ANPs, then you might need to recreate all of them to accommodate any changes in the future.
+
+* When using `0.0.0.0/0` or `::/0` to create a strong `Deny` policy, it is recommended that you check to make sure that you have higher priority `Allow` or `Pass` rules for essential traffic.
+
+* Use `Allow` when you want to ensure that a connection is allowed no matter what. An `Allow` rule in an ANP means that the connection will always be allowed, and `NetworkPolicy` will be ignored.
+
+* Use `Pass` if you want to "not deny" the connection, and allow developers to decide for themselves whether to allow or deny the connection according to their own `NetworkPolicy` objects.`Pass` delegates policy decision to the `NetworkPolicy` layer.
+
+* Selectors across multiple rules should not overlap. This ensures that that the same IPs do not appear in multiple policies, which can cause performance and scale limitations.
+
+* Avoid using `namedPorts` in conjunction with `PortNumber` and `PortRange` creates 6 ACLs and cause inefficiencies in your cluster.
+
+[id="considerations-using-banp_{context}"]
+== Considerations for using BaselineAdminNetworkPolicy
+
+* Only a single `BaselineAdminNetworkPolicy` (BANP) resource may be defined within a cluster, make sure you architect it thoughtfully. The following are supported uses for BANP that administrators might consider in designing their BANP:
+
+** You can set a default deny policy for cluster-local ingress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the ingress traffic that they want to allow, and if they do not add network policies for ingress it will be denied.
+
+** You can set a default deny policy for cluster-local egress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the egress traffic that they want to allow, and if they do not add network policies it will be denied.
+
+** You can set a default allow policy for egress to the in-cluster DNS service. Such a BANP will ensure that namespaced users do not have to set an allow egress `NetworkPolicy` to the in-cluster DNS service.
```

Comment:
I am not sure if `default allow policy` is a valid term. may be allow needs to be in backticks or the sentence s needs to use it like a verb? Same applies to deny.

```suggestion
** You can set a default allow policy for egress to the in-cluster DNS service. Such a BANP ensures that the namespaced users do not have to set an allow egress `NetworkPolicy` to the in-cluster DNS service.
```